openedx/core/djangoapps/catalog/utils.py

```diff
@@ -31,8 +45,11 @@ def get_programs(user, uuid=None, type=None):  # pylint: disable=redefined-built
         dict, if a specific program is requested.
     """
     catalog_integration = CatalogIntegration.current()
-
     if catalog_integration.enabled:
+        user = _get_service_user(user, catalog_integration.service_username)
+        if not user:
+            return []
```

Comment:
Looks like we need to cover this line with a test. https://build.testeng.edx.org/job/edx-platform-python-unittests-pr/29174/Diff_Coverage_Report/